zend_abstract_interface/config/config_decode.c

```diff
@@ -105,46 +105,78 @@ static bool zai_config_decode_int(zai_str value, zval *decoded_value) {
+
+                while (*data && *data != ',') {
                     if (*data == ':') {
-                        while (*++data && (*data == ' ' || *data == '\t' || *data == '\n'))
-                            ;
-
-                        value_start = value_end = data;
-                        if (!*data || *data == ',') {
-                            --value_end;  // empty string instead of single char
-                        } else {
-                            while (*++data && *data != ',') {
-                                if (*data != ' ' && *data != '\t' && *data != '\n') {
-                                    value_end = data;
-                                }
+                        has_colon = true;
+    
+                        data++;
+                        while (*data == ' ' || *data == '\t' || *data == '\n') data++;
+
+                        value_start = data;
+                        value_end = value_start;
+
+                        while (*data && *data != ',') {
+                            if (*data != ' ' && *data != '\t' && *data != '\n') {
+                                value_end = data;
                             }
+                            data++;
                         }
 
                         size_t key_len = key_end - key_start + 1;
                         size_t value_len = value_end - value_start + 1;
-                        zval val;
-                        ZVAL_NEW_STR(&val, zend_string_init(value_start, value_len, persistent));
-                        zend_hash_str_update(Z_ARRVAL(tmp), key_start, key_len, &val);
+
+                        if (key_len > 0 && value_len > 0) {
+                            zend_string *key = zend_string_init(key_start, key_len, persistent);
+                            if (lowercase) {
+                                zend_str_tolower(ZSTR_VAL(key), ZSTR_LEN(key));
+                            }
+
+                            zval val;
+                            ZVAL_NEW_STR(&val, zend_string_init(value_start, value_len, persistent));
+                            zend_hash_update(Z_ARRVAL(tmp), key, &val);
+                            zend_string_release(key);
+                        }
                         break;
                     }
+
                     if (*data != ' ' && *data != '\t' && *data != '\n') {
                         key_end = data;
                     }
+                    data++;
+                }
+
+                // Handle case without a colon (standalone key)
+                if (!has_colon && key_end) {
+                    size_t key_len = key_end - key_start + 1;
+                    zend_string *key = zend_string_init(key_start, key_len, persistent);
+                    if (lowercase) {
+                        zend_str_tolower(ZSTR_VAL(key), ZSTR_LEN(key));
+                    }
+
+                    zval val;
+                    ZVAL_NEW_STR(&val, zend_string_copy(key));
```

Comment:
wouldn't ZVAL_EMPTY_STR() make more sense here? As in: no value = empty value? Instead of no value = the key again?

Reply:
Yeah that makes sense I wasn't sure if use the key as value to make use of the value in the serializer or empty, will go for the later as it does make more sense and seems to be better performance wise.